### ISSUE TITLE
Show different images different days 

### ISSUE THREAD

**Issue by jopper1988** [xibosignage/xibo#346]
I want to show images of the employees that are working that day. Is there a possibility to do this ?

**Comment by MedusaDelft** [comment]
I suggest you create an RSS feed

**Comment by jopper1988** [comment]
Can i display images in a rss feed ?

**Comment by jopper1988** [comment]
And is there a tutorial to make a rss feed?  that will display images a and b on monday and c and d on tuesday. 

???

**Comment by Rating** [comment]
You can use https://picasaweb.google.com/ to show images from a RSS.

**Comment by MedusaDelft** [comment]
I suppose you have some digital registration of who is in or not. So that information is what you need to create an RSS feed. Otherwise, it is not possible without human intervention anyway (people get ill/sick, holidays, taking a course etc).

**Comment by jopper1988** [comment]
No there is no digital registration. Only a form which is filled when someone has to work. So i have to adjust it myself when someone get ill or is on a holiday. 

But we have the same persons on different days. For example person a and b working on monday, tuesday, wednesday, friday. 

Persons c and d working on monday, thursday, friday. 

We have a HRM system but that can't connect with xibo.

**Comment by MedusaDelft** [comment]
Maybe you can use a dataset, add a column for the days they work and filter the dataset using MySQL (see manual). I'm not sure it datasets allow images (link or blob). I'm pretty new to Xibo and have not really looked into datasets yet.

**Comment by jopper1988** [comment]
never worked with datasets but i will give it a try.

**Comment by mattgrimley** [comment]
I use a dataset to do something with images.. It's not the same thing, but the process might be helpful.. It is complicated, but really powerful!

Essentially, my dataset contains a bunch of times of the day "Period 1", "Registration", "Lunch Time" etc. Each entry contains the nice name, the start-time, the end-time and a URL for an image.

I use a filter on my ticker to extract the relevent data (for me, this looks at the current time and identifies which period we should be in). I then use the image URL as part of an <img> tag. so i get a bawdy flashing "1" during period 1 and a lunchbox pic during lunch.

You could do something similar to return all "People" who are currently set "active".. And each person would have a "Nice name" and an image URL returned.

**Comment by dasgarner** [comment]
Great suggestions - i'd also recommend a dataset - the Image Data type was added specifically for this purpose.

**Comment by jopper1988** [comment]
Ok then i will look into this one.

**Comment by jopper1988** [comment]
But how can i filter on days ? So on monday the images a and b will show and on tuesday it will show images c and d

**Comment by mattgrimley** [comment]
Well, a slightly clunky way of doing it would be to have a field for each day of the week with a 1 or 0 in it. Then you'd need to create an SQL filter to to retrieve only staff who have a 1 for the day which equals today. There is not always a clear right or wrong for database design or SQL queries.. Try to keep it simple and inevitably you'll think of a better way of doing it just after you finish your first attempt!! :)

As an example, in my case we use the following "Filter":
```
CURTIME() > TIME(dtStartTime) AND CURTIME() < TIME(dtEndTime)
```
This returns records if the "Current Time" is greater than the my field "Start-Time" AND less than my field "End-Time".

Hope this get's you on the right track! :)

**Comment by jopper1988** [comment]
I will give it a try but i never worked with Datasets before. It will be a great puzzle to solve ;-). But if i have it correct i need to make colums: Mondag to Friday ? But how do i get a 0 or 1 into it ?

**Comment by jopper1988** [comment]
What is the SQL code to filter the days ?

**Comment by mattgrimley** [comment]
I'm afraid, this is quite difficult and i'm no expert in SQL.. though, there is plenty of help out there.. The first thing you need to learn is how to run queries manually against your SQL database, so you can prepare and test your query. Honestly, i don't know how to do that, but you can google it as quickly as me ;)

Next, you need to write down your query in plain english. Your database design will ultimately get refined as you work out what is easy and what is hard.. for example, my suggestion of a column for each day is probably a really bad idea because it makes searching more complex..
There is a LIKE operator (http://www.w3schools.com/sql/sql_LIKE.asp) in SQL which might enable you to search a single column for a specific word.. so, perhaps you have a column which just lists the days that someone works "monday,thursday,friday".

So, your plain english query might be:
return all rows where column "Day" includes monday

The next challenge is to identify the "today" variable.. I believe that DATEPART (http://www.w3schools.com/sql/func_datepart.asp) is your friend here..
Something like this should give you the current weekday:
```
DATEPART(dw, CurDate())
```
so, your filter would look something like
```
DATEPART(dw, CurDate()) LIKE 'columnE'
```

Now, i'm fairly certain that datepart example isn't right, but i hope it'll get you on the right track (or perhaps someone with better sQl skills can correct it..

Sorry if my version of help has the opposite effect!!!

Matt

**Comment by MedusaDelft** [comment]
Assuming you use MySQL, it is: CURDATE()  for the current date and WEEKDAY for the day of the week (0=Monday etc.)  I would use a string column that contains a 0 if one works on monday, etc.  So having one person working on Monday, Tuesday and Thursday, it would contain 013. Working only in weekends, it would contain 56. I would use the INSTR function to validate this column. 

Something like:

    WHERE INSTR(column_with_days_as_numbers, WEEKDAY(CURDATE()) <> 0

The not 0-values garantees the weekday-number exists in the column.

See the MySQL site for more information (http://dev.mysql.com/). And good luck!

**Comment by dasgarner** [comment]
I might have misunderstood, but I believe the op wants something quite simple

> So on monday the images a and b will show and on tuesday it will show images c and d

In that case s/he could have a column called "Day" which contained a string which was either "Monday", "Tuesday", etc

Then the filter could be WEEK(CURDATE()) = col_with_day_name

Although I really like @MedusaDelft solution for multiple days! :+1:

**Comment by MedusaDelft** [comment]
@dasgarner my solution should only return all the rows for the current day. I think this is a more flexible solution in case you need to return the same rows (aka image(s)) on more than one day.

**Comment by dasgarner** [comment]
I completely agree with you - I just thought it might be a little easier to understand putting in Monday instead of 1 (or is that 0?)

**Comment by MedusaDelft** [comment]
Ah, I see your point.  Just thinking about storage space and posisble typo's    ;-)

**Comment by jopper1988** [comment]
I never used it before, i don't understand it what i need to do. Behind my xibo i run a webserver XAMPP. 

But what do i need to do? I understand that i need a column in Xibo with the title Day with the value 0,1,2 ???

And how do i filter it in Dataset ?

**Comment by mattgrimley** [comment]
Hey @jopper1988 - have a look at http://xibo.org.uk/2014/05/27/using-datasets-to-display-meeting-room-bookings/ for the basics of datasets - it may be tailored for the older version, but the principles are the same. That should get you started with the terminology etc :)

**Comment by dasgarner** [comment]
[This](http://xibo.org.uk/2013/02/20/howto-video-series-datasets/) is very much out of date, but may also provide a gentle introduction.
